make-vendored-tarfile.sh

```diff
@@ -1,14 +1,31 @@
-#/bin/bash
+#! /bin/bash
+
 set -x
-ver=$1
-cargo vendor
-# Various vendor cleanups
-pushd vendor
-# cleanup windows files
-rm -rf winapi/src/*
-touch winapi/src/lib.rs
-rm -rf winapi-x86_64-pc-windows-gnu/lib/*
-rm -rf winapi-i686-pc-windows-gnu/lib/*
-rm -rf vcpkg/test-data
-popd #vendor
-tar cJf fido-device-onboard-rs-$ver-vendor-patched.tar.xz vendor/
+VER=${1:-$(git rev-parse HEAD)}
```

Comment:
Does this still allow us to pass say "0.5.0" for an official release? This script is also used to generate the vendored tar files for official releases as well where we need to tag it with the same rev as the official release.

Reply:
Other than this query it LGTM.

Reply:
Yeah, the code after `:-` is the default value in case the variable before `:-` does not contain any value. The way the `make-vendored-tarfile.sh` script can be invoked is:
```
make-vendored-tarfile.sh [ VER [PLATFORM [PLATFORM [PLATFORM...]]]]
```
In case no arguments are passed the VER will be the result of executing `git rev-parse HEAD` and in case no PLATFORM is passed the platforms will be the result of executing `echo {x86_64,aarch64,powerpc64le,s390x}-unknown-linux-gnu`